core/src/main/java/edu/ucr/cs/riple/core/Annotator.java

```diff
@@ -96,6 +96,14 @@ private void explore() {
     Utility.buildTarget(config);
     Utility.setScannerCheckerActivation(config.target, false);
     FieldDeclarationAnalysis fieldDeclarationAnalysis = new FieldDeclarationAnalysis(config.target);
+    MethodInheritanceTree tree =
```

Comment:
Do we have any test for this `explore` method? Like i think it has like important logic that needs to have some tests around it.

Reply:
Presumably it gets exercised as part of the unit/integration [tests](https://github.com/nimakarimipour/NullAwayAnnotator/blob/master/core/src/test/java/edu/ucr/cs/riple/core/DownstreamAnalysisTest.java#L32) for the full downstream dependency analysis? Or are you thinking of a more targeted way to test `explore()` in isolation? (Presumably mocking the actual compilation task results?)

Reply:
I was wondering if there are any E2E test cases that are affected due to this change.  Does the change to the `.tsv` file update the E2E test case? 

Also in general, I saw that this `explorer` method is calling a bunch of APIs -> `ExaustiveExplorer`, `OptimisedExplorer`,`Bank` .. and it is doing a lot of things. Mocking and testing all of this would be too much effort (unless there is a way of smartly extracting out the core exploration functionality and testing it in isolation), 
. At least for now, It would be either helpful have a a method level comment describing the functionality of this method or line-by-line comments .

Reply:
@lazaroclapp @ketkarameya Sure I agree that it needs a unit test to check functionality of `explorers` in a finer granularity level and you are right it needs a significant effort to mock this. The main reason for that is `explorers` require a command to re-run the analysis (NullAway at this moment) and mocking that requires a lot of work. But I am definitely interested in working on this in future.

Reply:
Correct, it is unrelated and also exactly correct about the explanation why this does not change within iterations.